SunEngine/StartupConfiguration.cs

```diff
@@ -11,14 +11,50 @@ namespace SunEngine
     public class StartupConfiguration
     {
         private static readonly Logger Logger = LogManager.GetCurrentClassLogger();
-
+        
         private const string ConfigurationArgumentName = "config:";
         private const string DefaultConfigurationFileName = "Config";
 
+        private const string HelpCommand = "help";
+        private const string VersionCommand = "version";
+        private const string ServerCommand = "server";
+        private const string MigrateCommand = "migrate";
+        private const string InitCommand = "init";
+        private const string SeedCommand = "seed";
+        private const string AppendCategoriesNamesCommand = "append-cat-name";
+
+        public string[] Arguments { get; }
         public string ConfigurationDirectoryRoute { get; }
+        public bool PrintHelp { get; }
+        public bool PrintVersion { get; }
+        public bool StartServer { get; }
 
-        public StartupConfiguration(IEnumerable<string> arguments)
+        public bool Migrate { get; }
+        public bool InitializeCoreData { get; }
+        
+        public bool SeedWithTestData { get; }
+        public bool SeedWithCategoryNames { get; }
+        public List<string> CategoryTokensToSeed { get; }
+
+        public StartupConfiguration(string[] arguments)
         {
+            Arguments = arguments;
+            
+            var startupArguments = arguments.ToHashSet();
+
+            PrintHelp = startupArguments.Contains(HelpCommand);
+            PrintVersion = startupArguments.Contains(VersionCommand);
+            
+            StartServer = startupArguments.Contains(ServerCommand);
+            InitializeCoreData = startupArguments.Contains(InitCommand);
+            Migrate = startupArguments.Contains(MigrateCommand);
+            
+            SeedWithTestData = startupArguments.Contains(SeedCommand);
+            SeedWithCategoryNames = startupArguments.Contains(AppendCategoriesNamesCommand);
+            
+            // TODO Revisit logic of category tokens. It could be seed:catToken, seed-catToken etc. Write logic to contain only category names, without "seed"
+            CategoryTokensToSeed = startupArguments.Where(x => x.StartsWith(SeedCommand)).ToList();
```

Comment:
We need to provide category names or tokens to the migrator/seeder. Logic should be improved here, but it could be tricky because use cases aren't clear.

Reply:
Yes. Not good to parse in Seeder project. The best way to move seed: params to be parsed in separate SunEngine.dll method/class

Reply:
Need to make Seeder Config class to push Seed parameters by this class